include/extractor/guidance/intersection_handler.hpp

```diff
@@ -508,15 +508,15 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
             // even reverse the direction. Since we don't want to compute actual turns but simply
             // try to find whether there is a turn going to the opposite direction of our obvious
             // turn, this should be alright.
-            NodeID new_node;
-            const auto previous_intersection = [&]() {
-                EdgeID turn_edge;
-                std::tie(new_node, turn_edge) = intersection_generator.SkipDegreeTwoNodes(
+            const auto previous_intersection = [&]() -> IntersectionView {
+                const auto parameters = intersection_generator.SkipDegreeTwoNodes(
```

Comment:
Hm. I could potentially think of implementing a specialised graph walker. The skipping itself does not accumulate and can only handle degree two intersections. It's more efficient than the graph walking process that needs to generate intersections/bearings. What would be your take here? I'd also like to unify it, but the only thing I could think of is to split the skipping from the intersection_generator. It is tightly coupled to its usages to intersection generation, though.